front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/ItemDAO.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.front50.model
+
+import com.netflix.spinnaker.front50.exception.NotFoundException
+
+interface ItemDAO<T> {
```

Comment:
How's this look @ttomsu?

create()/update() seem a little strange to me but seem to fit at least the majority of the previous DAO implementations. 

I may still revisit the need for a `String id` parameter once I have the controller TCK tests completed.